be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/Validator.scala

```diff
@@ -28,29 +28,29 @@ object Validator {
 
   private def validateMethodContent(graphMessage: GraphMessage): GraphMessage = graphMessage match {
     case Right(jsonRpcRequest: JsonRpcRequest) => jsonRpcRequest.getParams match {
-        case _: Broadcast   => validateBroadcast(jsonRpcRequest)
-        case _: Catchup     => validateCatchup(jsonRpcRequest)
-        case _: Publish     => validatePublish(jsonRpcRequest)
-        case _: Subscribe   => validateSubscribe(jsonRpcRequest)
-        case _: Unsubscribe => validateUnsubscribe(jsonRpcRequest)
-        case _              => Left(validationError(jsonRpcRequest.id))
+        case _: Broadcast       => validateBroadcast(jsonRpcRequest)
+        case _: Catchup         => validateCatchup(jsonRpcRequest)
+        case _: Publish         => validatePublish(jsonRpcRequest)
+        case _: Subscribe       => validateSubscribe(jsonRpcRequest)
+        case _: Unsubscribe     => validateUnsubscribe(jsonRpcRequest)
+        case _: Heartbeat       => graphMessage // No check necessary
+        case _: GetMessagesById => graphMessage // No check necessary
+        case _                  => Left(validationError(jsonRpcRequest.id))
       }
-    case Right(jsonRpcResponse: JsonRpcResponse) => Left(PipelineError(
-        ErrorCodes.SERVER_ERROR.id,
-        "Unsupported action: MethodValidator was given a response message",
-        jsonRpcResponse.id
-      ))
```

Comment:
Was it necessary to remove that?

Reply:
get_messages_by_id responses wouldn't get through otherwise. The json answers our backend is able to parse is fairly limited, basically: `result: 0` + catchup (list of messages) + now the maps channels -> list(messages)
So if the answer is outside of these 3, it can't parse it and an error is thrown. Finally if it is one of these 3 then `GetMessagesByIdResponseHandler` will deal with it, either silently dropping them or treating the maps.

Reply:
Ok thanks for the explanation, it indeed makes sense not to reject them now that we can receive responses from other servers!
However, I believe it would be clearer to keep the explicit case handling and always accept it (as you did for `Heartbeat` and `GetMessagesById`), both for `validateMethodContent` and `validateMessageContent`. In fact, I don't see the point of validating messages that are not explicitly handled, I would find it safer to reject messages if they don't match the expected patterns (which would then force us to explicitly handle jsonRpcResponse messages, even if trivially).

Reply:
I couldn't add the comments under the relevant lines as they were not affected by these commits, so here are the comments I wanted to add to the previous idea:
- line 41 of Validator.scala: `case _ => graphMessage`
- line 55 of Validator.scala: `case graphMessage @ _ => graphMessage`
As explained above, for safety reasons shouldn't they be rejected (`Left(...)`) if they are not handled? (It's not what your commit exactly is about, but this could be a good moment to change that if needed)

Reply:
These are all good ideas, accept by default are not great, far from it. But it starts to introduce a lot in the same pr, I feel like this should be in its own refactoring work in order to avoid tangling together too many changes. What do you think ?

Reply:
You're right, I'll do an issue about it and a simple pr should solve it !